test/ref/topk.cpp

```
@@ -30,54 +30,96 @@
+        std::iota(indices.rbegin(), indices.rend(), 1);
+        topk_inputs.push_back(mm->add_literal(migraphx::literal{is, indices}));
+    }
+    auto r  = mm->add_instruction(migraphx::make_op("topk", op), topk_inputs);
+    auto r0 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 0}}), r);
+    auto r1 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 1}}), r);
+    mm->add_return({r0, r1});
+
+    p.compile(migraphx::make_target("ref"));
 
-        return p;
+    std::vector<float> input_data = {
+        2.1,
+        2.3,
+        2.0,
+        2.5,
+        1.9,
+        3.3,
+        0.2,
+        4.5,
+        0.1,
+        0.8,
+        1.0,
+        4.5,
+        2.1,
+        0.8,
+        1.5,
     };
+    migraphx::parameter_map pp;
+    pp["data"] = migraphx::argument(s, input_data.data());
+    auto rets  = p.eval(pp);
+    std::vector<float> ret_val;
+    rets.front().visit([&](auto v) { ret_val.assign(v.begin(), v.end()); });
+    std::vector<int64_t> ret_ind;
+    rets.back().visit([&](auto v) { ret_ind.assign(v.begin(), v.end()); });
 
-    auto run_program = [&](int64_t k, int64_t axis, int largest) {
-        auto p = create_program(k, axis, largest);
-        p.compile(migraphx::make_target("ref"));
-        std::vector<float> data = {
-            2.1, 2.3, 2.0, 2.5, 1.9, 3.3, 0.2, 4.5, 0.1, 0.8, 1.0, 4.5, 2.1, 0.8, 1.5};
-        migraphx::shape s{migraphx::shape::float_type, {3, 5}};
-        migraphx::parameter_map pp;
-        pp["data"] = migraphx::argument(s, data.data());
-        auto rets  = p.eval(pp);
-        std::vector<float> ret_val;
-        rets.front().visit([&](auto v) { ret_val.assign(v.begin(), v.end()); });
-        std::vector<int64_t> ret_ind;
-        rets.back().visit([&](auto v) { ret_ind.assign(v.begin(), v.end()); });
+    return std::make_pair(ret_val, ret_ind);
+}
 
-        return std::make_pair(ret_val, ret_ind);
-    };
+TEST_CASE(topk_largest0)
```

Please add tests for `sorted` `topk` here.

---

It always sorted.

---

Then, please throw an exception in onnx parsing if the topk operator attribute: `sorted == 0`.

---

The onnx parser is beyond the scope of this PR, but I dont think we should throw an exception. Being sorted is acceptable for onnx when `sorted == 0` because the data can be in any order(which also includes sorted order).